src/txmempool.cpp

```diff
@@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
     BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
         mapNextTx.erase(txin.prevout);
 
+    if (vTxHashes.size() > 1) {
+        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() - 1];
+        txiter newhashit = mapTx.find(vTxHashes[it->vTxHashesIdx].first);
```

Comment:
If not for this find, we wouldn't need a wtxid index in the mempool... as long we don't build vTxHashes asynchronously, it could be done by storing a CTxMemPool::txiter in vTxHashes instead.